File: leetcode/1704. Determine if String Halves Are Alike.py
class Solution:
    def halvesAreAlike(self, s: str) -> bool:
        s = s.lower()
        vowel_letter = "aeiou"
        n = len(s)
        half = n // 2
        count = 0
        
        for i in range(0, half):
            if s[i] in vowel_letter:
                count += 1

        for i in range(half, len(s)):
            if s[i] in vowel_letter:
                count -= 1

        return count == 0

File: leetcode/1704. Determine if String Halves Are Alike.cpp
class Solution {
public:
    bool halvesAreAlike(string s) {
        int n = s.length(), half = n / 2, count = 0;
        for(int i = 0; i < half; ++i){
            if(isVowelLetter(s[i])){
                ++count;
            }
        }
        for(int i = half; i < n; ++i){
            if(isVowelLetter(s[i])){
                --count;
            }
        }
        return count == 0;
    }

    bool isVowelLetter(char c){
        c = (char)((int)c | 32); // change uppercase to lowercase 
        if(c == 'a') return true; 
        if(c == 'e') return true; 
        if(c == 'i') return true; 
        if(c == 'o') return true;
        if(c == 'u') return true; 
        return false;
    }
};